File: app/src/main/java/at/tomtasche/reader/background/BillingPreferences.java
package at.tomtasche.reader.background;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.SharedPreferences.Editor;

import at.tomtasche.reader.BuildConfig;

public class BillingPreferences {

    private final SharedPreferences sharedPreferences;

    public BillingPreferences(Context context) {
        sharedPreferences = context.getSharedPreferences(
                "modifyMeIfYouWantToRemoveAdsIllegally", Context.MODE_PRIVATE);
    }

    public boolean hasPurchased() {
        if (sharedPreferences == null) {
            return false;
        }

        return sharedPreferences.getBoolean("purchaseAcknowledged", false);
    }

    public void setPurchased(boolean purchased) {
        if (sharedPreferences == null) {
            return;
        }

        Editor editor = sharedPreferences.edit();
        editor.putBoolean("purchaseAcknowledged", purchased);
        editor.apply();
    }
}

File: app/src/main/java/app/opendocument/droid/background/BillingPreferences.kt
package app.opendocument.droid.background

import android.content.Context
import androidx.core.content.edit

class BillingPreferences(context: Context) {

    private val sharedPreferences =
        context.getSharedPreferences(
            "modifyMeIfYouWantToRemoveAdsIllegally",
            Context.MODE_PRIVATE,
        )

    fun hasPurchased(): Boolean = sharedPreferences.getBoolean(PURCHASE_KEY, false)

    fun setPurchased(purchased: Boolean) {
        sharedPreferences.edit { putBoolean(PURCHASE_KEY, purchased) }
    }

    private companion object {
        const val PURCHASE_KEY = "purchaseAcknowledged"
    }
}
